File: FinalProject/app/src/main/java/com/example/finalproject/Armor.java
package com.example.finalproject;


public class Armor extends Item {
    private int armor, weight, type_of_armor;

    public Armor(int armor, int weight, int type_of_armor, int rarity) {
        this.armor = armor;
        this.weight = weight;
        this.type_of_armor = type_of_armor;
        super.setRarity(rarity);
    }

    public int getArmor() {
        return armor;
    }

    public void setArmor(int armor) {
        this.armor = armor;
    }

    public int getWeight() {
        return weight;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }

    public int getType_of_armor() {
        return type_of_armor;
    }

    public void setType_of_armor(int type_of_armor) {
        this.type_of_armor = type_of_armor;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/items/Armor.kt
package com.example.finalproject.service.classes.items

class Armor(
    name: String,
    id: Int,
    costSell: Int,
    costBuy: Int,
    category: Int,
    rarity: Int,
    val armor: Double,
    val weight: Double,
    val typeOfArmor: Int
) : Item(
    name = name,
    id = id,
    costSell = costSell,
    costBuy = costBuy,
    rarity = rarity,
    category = category
)
